base/scripts/objectives.cs

```diff
@@ -466,8 +476,27 @@
 
 function Player::enterMissionArea(%this)
 {
+    
  %set = nameToID("MissionCleanup/ObjectivesSet");
  %this.outArea = "";
+ 
+ %playerClient = Player::getClient(%this);
+ %playerTeam = GameBase::getTeam(%this);
+ %ClientName = Client::getName(%playerClient);
+    
+    if (%playerTeam == 0) {
+        %flagTeam = 1;
+    }
+    else {
+       %flagTeam = 0; 
+    }
```

Comment:
this is more concise and also looks like you don't use ClientName anymore:
```suggestion
 %playerClient = Player::getClient(%this);
 %playerTeam = GameBase::getTeam(%this);
 %flagTeam = (%playerTeam + 1) % 2;
```